src/Command/ErrorFormatter/SarifErrorFormatter.php

```diff
@@ -0,0 +1,118 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use Bartlett\Sarif\Definition\ArtifactLocation;
+use Bartlett\Sarif\Definition\Location;
+use Bartlett\Sarif\Definition\Message;
+use Bartlett\Sarif\Definition\PhysicalLocation;
+use Bartlett\Sarif\Definition\PropertyBag;
+use Bartlett\Sarif\Definition\Region;
+use Bartlett\Sarif\Definition\Result;
+use Bartlett\Sarif\Definition\Run;
+use Bartlett\Sarif\Definition\Tool;
+use Bartlett\Sarif\Definition\ToolComponent;
+use Bartlett\Sarif\SarifLog;
+
+use PHPStan\Command\AnalysisResult;
+use PHPStan\Command\Output;
+
+/**
+ * @author Laurent Laville
+ */
+class SarifErrorFormatter implements ErrorFormatter
+{
+
+	public function formatErrors(AnalysisResult $analysisResult, Output $output): int
+	{
+		$driver = new ToolComponent('PHPStan');
+		$driver->setInformationUri('https://phpstan.org');
+		$driver->setVersion('1.1.2');
```

Comment:
The version can be obtained like this: https://github.com/phpstan/phpstan-src/blob/1789915a3ed766e8e784ac3eabee4b327e9d192a/bin/phpstan#L109-L114